docs/modules/tls.md

```diff
@@ -20,20 +20,29 @@ most common features supported.  Specifically, it provides:
 - key exchange algorithms: DHE and ECDHE
 - elliptic curves: secp{256,384}r1, secp256k1, bp{256,384}.
 
-!!! tip
-	If possible, you will likely be much better served by using the ECDSA
-	signature and key exchange algorithms than by using RSA.  An
-	increasingly large fraction of the Internet understands ECDSA, and most
-	server software can speak it as well.  The much smaller key size (at
-	equivalent security!) is beneficial for NodeMCU's limited RAM.
-
-	https://wiki.openssl.org/index.php/Command_Line_Elliptic_Curve_Operations
-	details how to create ECDSA keys and certificates.
+!!! warning
 
-!!! tip
-	The complete configuration is stored in [user_mbedtls.h](../../app/include/user_mbedtls.h). This is the file to edit if you build your own firmware and want to change mbed TLS behavior.
+	The severe memory constraints of the ESP8266 mean that the `tls` module
+	is by far better suited for communication with custom, purpose-built
+	endpoints with small certificate chains (ideally, even self-signed)
+	than it is with the Internet at large.  By default, our mbedTLS
+	configuration requests TLS fragments of at most 4KiB and is unwilling
+	to process fragmented messages, meaning that the entire ServerHello,
+	which includes the server's certificate chain, must conform to this
+	limit.  We do not believe it useful or easy to be fully compliant with
+	the TLS specification, which requires a 16KiB recieve buffer and,
+	therefore, 32KiB of heap within mbedTLS.  While it is possible to
+	slightly raise the buffer sizes with custom nodeMCU builds, connecting
+	to endpoints out of your control will remain a precaious position, and
+	so we strongly suggest that TLS connections be made only to endpoints
+	under your control, whose TLS configurations can ensure that their
+	ServerHello messages are small.  A reasonable compromise is to have a
+	"real" computer do TLS proxying for you; the `socat` program is one
```

Comment:
Linking to https://medium.com/@copyconstruct/socat-29453e9fc8a6 or the man page at https://linux.die.net/man/1/socat ?